sms/src/main/java/com/gruppo4/sms/network/SMSAbstractNetworkManager.java

```diff
@@ -0,0 +1,212 @@
+package com.gruppo4.sms.network;
+
+import com.gruppo4.communication.network.NetworkManager;
+import com.gruppo4.communication.network.SerializableObject;
+import com.gruppo4.sms.dataLink.SMSHandler;
+import com.gruppo4.sms.dataLink.SMSMessage;
+import com.gruppo4.sms.dataLink.SMSPeer;
+
+import java.util.ArrayList;
+
+/**
+ * This class is intended to be extended by the specific application. It is an implementation of NetworkManager.
+ * Each SMS request has a format specified below. A node cannot inform each node in the network about his changes to
+ * the local dictionary, for it will be expensive in terms of messages sent. Thus we ordered our dictionary by phone number, so that
+ * each node has the same dictionary ordered in the same way: a node with index n in the dictionary will spread changes to only nodes numbered
+ * 2n+1, 2n+2, guaranteeing mutual exclusion.
+ * This costs 2 messages for each node and delivery of information is logarithmic in N, the extension of the network.
+ *
+ * @author Marco Mariotto
+ */
+
+public abstract class SMSAbstractNetworkManager implements NetworkManager<SMSPeer, SerializableObject, SerializableObject>
+{
+    /*
+     * SMS REQUESTS FORMATS
+     * Join proposal:    "JP_%netName"
+     * Add user:         "AU_%(peer)"          we include the whole peer, not only his address
+     * Remove user:      "RU_%(address)"       address is the phone number of the user being removed
+     * Add resource:     "AR_%(key)_%(value)"  we include the whole resource, key and value
+     * Remove resource:  "RR_%(key)"           we only need the key to identify a resource
+     * Don't spread:     "%(1)DS_%(2)"         inform the receiver to not spread this info, it's one of the previous formats where %1 != JP
+     */
+
+    static final String ADD_USER = "AU";
+    static final String REMOVE_USER = "RU";
+    static final String ADD_RESOURCE = "AR";
+    static final String REMOVE_RESOURCE = "RR";
+    static final String DO_NOT_SPREAD = "DS";
+    static final String JOIN_AGREED = "JA";
+    static final String JOIN_PROPOSAL = "JP";
+    static final String SPLIT_CHAR = "_";
+
+    private SMSNetworkDictionary dict = new SMSNetworkDictionary<>();
+
+    // joinSent keeps track of JOIN_PROPOSAL requests still pending.
+    private ArrayList<SMSPeer> joinSent = new ArrayList<>();
+    protected  String networkName;
+    // mySelf is the current peer setting up the network
+    protected  SMSPeer mySelf;
+
+    //manager makes use of SMSHandler to send requests
+    private SMSHandler handler;
+
+    /**
+     * Set up a new network
+     *
+     * @param handler we set up a handler for sending requests
+     * @param networkName of the network being created
+     */
+    public void setup(SMSHandler handler, String networkName, SMSPeer mySelf){
+        this.handler = handler;
+        this.networkName = networkName;
+        this.mySelf = mySelf;
+        dict.addUser(mySelf);
+    }
```

Comment:
Since `SMSAbstractNetworkManager` is not a singleton, how do you manage multiple objects of the same Network Manager? Since they're different objects, do they get updated in the same way or only one of them?